app/controllers/foreman_ansible/api/v2/hostgroups_controller_extensions.rb

```diff
@@ -64,6 +65,28 @@ def assign_ansible_roles
             find_resource
             process_response @hostgroup.update(:ansible_roles => @ansible_roles)
           end
+
+          api :POST, '/hostgroups/:id/ansible_roles',
+              N_('Directly add an Ansible role to a hostgroup')
+          param :id, :identifier, :required => true
+          param :ansible_role_id, :identifier,
+                N_('Ansible role to add to a hostgroup'),
+                :required => true
+
+          def add_ansible_role
+            process_response @hostgroup.ansible_roles << @ansible_role
```

Comment:
What if the role is already in there? Is `ansible_roles` a set or a list?

Reply:
If the role is already assigned, it will return
```JSON
{
    "error": {
        "message": "Validation failed: Ansible role has already been taken"
    }
}
```
with `500` error code.

Reply:
I agree that `500` code is weird for such response, but Rails will treat this as a validation error anyway. I left `422` instead of suppressing the error. If you still find this wrong, I can change it to suppress the `ActiveRecord::RecordInvalid` and return `200`.